great_expectations/validation_operators/types/validation_operator_result.py

```diff
@@ -0,0 +1,258 @@
+        if self._actions_results_by_validation_result_identifier is None:
+            self._actions_results_by_validation_result_identifier = {
+                validation_result_identifier: run_result["actions_results"]
+                for (
+                    validation_result_identifier,
+                    run_result,
+                ) in self.run_results.items()
+            }
+        return self._actions_results_by_validation_result_identifier
+
+    @property
+    def expectation_suite_names(self) -> List[str]:
+        if self._expectation_suite_names is None:
+            self._expectation_suite_names = list(
+                set(
+                    [
+                        validation_result_identifier.expectation_suite_identifier.expectation_suite_name
+                        for validation_result_identifier in self._run_results.keys()
+                    ]
+                )
+            )
+        return self._expectation_suite_names
+
+    @property
+    def validation_result_identifiers(self) -> List[ValidationResultIdentifier]:
+        if self._validation_result_identifiers is None:
+            self._validation_result_identifiers = list(self._run_results.keys())
+        return self._validation_result_identifiers
+
+    @property
+    def run_results(
+        self,
+    ) -> Dict[
+        ValidationResultIdentifier,
+        Dict[str, Union[ExpectationSuiteValidationResult, dict]],
+    ]:
+        return self._run_results
+
+    @property
+    def validation_results(self) -> List[ExpectationSuiteValidationResult]:
+        if self._validation_results is None:
+            self._validation_results = [
+                run_result["validation_result"]
+                for run_result in self.run_results.values()
+            ]
+        return self._validation_results
+
+    @property
+    def run_id(self) -> str:
+        return self._run_id
+
+    @property
+    def evaluation_parameters(self) -> Union[dict, None]:
+        return self._evaluation_parameters
+
+    @property
+    def data_assets_validated(self) -> List[dict]:
+        if self._data_assets_validated is None:
+            self._data_assets_validated = list(
+                self.data_assets_validated_by_batch_id.values()
+            )
+        return self._data_assets_validated
+
+    @property
+    def data_assets_validated_by_batch_id(self) -> dict:
+        if self._data_assets_validated_by_batch_id is None:
+            assets_validated_by_batch_id = {}
+
+            for validation_result in self.validation_results:
+                batch_kwargs = validation_result.meta["batch_kwargs"]
+                batch_id = BatchKwargs(batch_kwargs).to_id()
+                expectation_suite_name = validation_result.meta[
+                    "expectation_suite_name"
+                ]
+                if batch_id not in assets_validated_by_batch_id:
+                    assets_validated_by_batch_id[batch_id] = {
```

Comment:
@Aylr - would this have helped in generating that client request (copied below)? Or perhaps one of: self._validation_results_by_expectation_suite_name, self._validation_results_by_data_asset_name

```
Quality Expectations, Profiles and Operational Check Details: https://s3.console.aws.amazon.com/s3/buckets/ge-data-documents
Please review the expectation failure summaries below:
- operational_check_1: Failed! 90.0 % passed (9 of 10 expectations were met)
- operational_check_2: Failed! 90.0 % passed (9 of 10 expectations were met)
- operational_check_3: Failed! 90.0 % passed (9 of 10 expectations were met)
- operational_check_4: Passed! 100.0 % passed (10 of 10 expectations were met)
```

Reply:
All as above

Reply:
Rob this would have helped tremendously! I like that there are multiple ways to get at this - via the suite names or via the data asset names. Nice!